jupyter_server/services/eventlog/handlers.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+from tornado import web
+
+from notebook.utils import url_path_join
+from notebook.base.handlers import APIHandler, json_errors
+from jupyter_telemetry.eventlog import EventLog
+
+class EventLoggingHandler(APIHandler):
+    """
+    A handler that receives and stores telemetry data from the client.
+    """
+    @json_errors
+    @web.authenticated
+    def post(self, *args, **kwargs):
+        try:
+            # Parse the data from the request body
+            raw_event = json.loads(self.request.body.strip().decode())
+        except Exception as e:
+            raise web.HTTPError(400, str(e))
+        
+        required_fields = {'schema', 'version', 'event'}
+        for rf in required_fields:
+            if rf not in raw_event:
+                raise web.HTTPError(400, '{} is a required field'.format(rf))
+
+        schema_name = raw_event['schema'] 
+        version = raw_event['version']
+        event = raw_event['event']
+        
+        # Profile, may need to move to a background thread if this is problematic
+        try: 
+            self.eventlog.record_event(schema_name, version, event)
```

Comment:
This requires the schema to be registered serverside before clients can post to it. How do we do that? Perhaps a traitlet, with corresponding entrypoint based loading as well?

Reply:
Ok let me create a separate PR for this.

Reply:
Pinged you on gitter about this but Imma just copy paste here.

This means something like a `jupyter telemetry [register|remove|list] <schema>` command?
register would put the schema file in a event-schemas directory. The directory structure would be something like `event-schemas/<name>/version/<schema>.[json|yaml]`

Reply:
That sounds great! I'm thinking that most things that want to submit events will be classic notebook extensions or jupyterlab extensions. These will be python packages, so perhaps we can begin by just allowing entrypoints? You call it, and it returns a bunch of schemas to register.

Reply:
Ah that makes more sense. I was thinking of another type of entrypoint. I'll make a separate PR on this one.